### DISCUSSION TITLE
How to run it concurrently in batch file

### DISCUSSION THREAD

**Discussion by megatron100** [charlesbel/Microsoft-Rewards-Farmer#311] [General]
I want to start multiple threads concurrently but I am getting issue with chrome. Is there any solid way to run concurrently

**Comment by N0TKZ** [comment]
what do you mean by "concurrently" are you asking how to run the command from the batch file or farming all account at the same time?

**Comment by aliasfoxkde** [comment]
@megatron100 It's an interesting thought and what I think you are looking for is "from concurrent import futures" which allows you to run multiple instances at the same time. I think it would only be practical (or manageable, and memory efficient) for headless mode, which might be more likely to get you banned. Also, just keep in mind that MS TOC's limit 5 accounts (a family) to a single IP address, so you would also need to implement a VPN connection for each set of accounts (if more than 5). Totally doable, but requires some development and programming knowledge. 

P.S. If I were to do this, I would only make each set (5) of accounts concurrent, each tied to a VPN session (needs to be US and probably consistent to the given group). Another thing to consider is you might have a problem with 2FA, but what I imagine might work is to just start a GUI session, login to each with 2FA validation, "save this device" which is unique to the Selenium browser session itself by User Agent, and you might be golden for ~90 days (but there might need to be a check that after X failed attempts, just stop/log as not to get banned, and notify you to re-authenticate or check for the issue).
